infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesPods.java

```diff
@@ -569,7 +575,15 @@ public void delete() throws InfrastructureException {
             watch.close();
           });
     } catch (KubernetesClientException ex) {
+      if (toCloseOnException != null) {
+        toCloseOnException.close();
+      }
       throw new KubernetesInfrastructureException(ex);
+    } catch (Exception e) {
+      if (toCloseOnException != null) {
+        toCloseOnException.close();
+      }
+      throw e;
```

Comment:
Maybe it would be better to wrap this exception into `InternalInfastructureException`

Reply:
well, honestly I prefer re-throwing exactly the same exception, so that there is no behavior change compared to the previous implementation that used to only catch `KubernetesClientException` exceptions.

Reply:
I agree that it's the same as was before. But only InfrastructureException is declared as throwing and it would be safer to wrap other exception in `InternalInfastructureException`.
`InternalInfastructureException` is special kind of exception that should wrap unexpected exceptions like NullPointerException which happen because of developer fault.
I would recommend wrapping but you can leave it as it is since it doesn't change current approach.